storage_controller/src/service.rs

```diff
@@ -6233,6 +6236,15 @@ impl Service {
         let (response, child_locations, waiters) =
             self.tenant_shard_split_commit_inmem(tenant_id, new_shard_count, new_stripe_size);
 
+        // Notify all page servers to detach and clean up the old shards because they will no longer
+        // be needed. This is best-effort: if it fails, it will be cleaned up by subsequent startup
+        // reconciliations.
```

Comment:
Hm. I don't think the start-up reconciliation part is true here. We don't have the parent tenant in the database anymore, so it will get detached when the pageserver re-attaches and the storage controller does not include te parent in the response.

Reply:
Yeah, pageserver restart will clean up. To me "startup reconciliation" is a specific stage in the storcon start-up (`Service::startup_reconcile`). I guess the pageserver meant pageserver start-up. Might be good to just say re-attach and avoid confusion.

Reply:
Ack, will update.